packages/component-library/src/components/plmg-radio-button-group/plmg-radio-button-group.tsx

```diff
@@ -0,0 +1,156 @@
+   *   - medium
+   *   - large
+   *
+   * Default: medium
+   */
+  @Prop() size: PlmgRadioButtonSize = 'medium';
+  @Watch('size')
+  validateSize(newValue: string) {
+    if (!isPlmgRadioButtonSize(newValue))
+      throw new Error('size: must be a valid value');
+  }
+
+  /**
+   * Define radio group's required status
+   *
+   * Allowed values:
+   *   - true
+   *   - false
+   *
+   * Default: false
+   */
+  @Prop() required: boolean = false;
+  @Watch('required')
+  validatesRequired(newValue: boolean) {
+    if (typeof newValue !== 'boolean')
+      throw new Error('required: must be boolean');
+  }
+
+  /**
+   * Define each radio button's value
+   *
+   * Accepts an array or JSON string
+   */
+  @Prop() values: string[] | string;
+  @State() parsedValues: string[];
+  @State() isValid: boolean;
+
+  @Watch('values')
+  parseValuesProp(newValue: string | string[]) {
+    if (newValue) {
+      this.parsedValues = Array.isArray(newValue)
+        ? newValue
+        : JSON.parse(newValue);
+    }
+  }
+  @Watch('values')
+  validatesValues(newValue: string[] | string) {
+    if (
+      typeof newValue !== 'object' ||
+      (typeof newValue === 'object' &&
+        !Array.isArray(newValue) &&
+        typeof newValue !== 'string')
+    ) {
+      throw new Error('values: must be array or string');
+    }
+    if (
+      Array.isArray(newValue) &&
+      newValue.some((value) => typeof value !== 'string')
+    ) {
+      throw new Error('values: values within array must be strings');
+    }
+  }
+
+  /**
+   * Define error message for radio group
+   * to be displayed if form validation
+   * fails
+   *
+   * Will render one error message for the
+   * radio button group, affects styling of
+   * all radio buttons in group
+   */
+  @Prop() errorMessage?: string | undefined;
+  @Watch('errorMessage')
+  validatesErrorMessage(newValue: string) {
+    if (newValue !== undefined && typeof newValue !== 'string')
+      throw new Error('errorMessage: must be a string if provided');
+  }
+
+  componentWillLoad() {
+    this.parseValuesProp(this.values);
+    this.isValid = true;
+  }
+
+  render() {
+    const legendClasses = {
+      large: this.size === 'large',
+    };
+
+    return (
+      <fieldset class={'plmg-radio-button-group'}>
+        <legend class={legendClasses}>
+          {this.label}
+          {this.required && <span>*</span>}
+        </legend>
+        {this.parsedValues.map((radio) => (
+          <plmg-radio-button
+            value={radio}
+            name={this.name}
+            size={this.size}
+            highlighted={!this.isValid}
+            isValid={(inputValid: boolean) => {
+              if (this.isValid !== inputValid) {
+                this.isValid = inputValid;
+              }
+            }}
+            required={this.required}
+          />
+        ))}
+
+        {!this.isValid && (
+          <plmg-error-message size={this.size} message={this.errorMessage} />
+        )}
```

Comment:
Before I thought about improving this, I wanted to hear your thoughts @JakeH91 @TanguyDucky 
In order to get the custom error component working, I've used `onInvalid` and `onInput` attributes on the radio button input component itself, together with this `isValid` prop here. When we submit the form and validation fails, `onInvalid` is called and it will run the function we pass in `isValid` (so we can change our `isValid` state and show the error message). `onInput`, we set `isValid` back to `true`. I had to use `preventDefault` in `onInvalid` to prevent the default browser error from showing up.

Is this gross?

Reply:
Lol I love the final question, it's something I often ask myself when submitting code lol. A lot of things in this PR are a bit gross, since the designers want things that go against default behaviour, so I don't think we have much choice other than to be gross.
![image](https://user-images.githubusercontent.com/13942249/171856536-1a00f771-42fa-4dd0-99fb-72f7ab08be82.png)

Reply:
Hi! I think this is quite smart and works perfectly :ok_hand: